PowerEditor/src/WinControls/ProjectPanel/TreeView.cpp

```diff
@@ -197,14 +220,15 @@ void TreeView::cleanSubEntries(HTREEITEM hTreeItem)
 {
 	for (HTREEITEM hItem = getChildFrom(hTreeItem); hItem != NULL; hItem = getNextSibling(hItem))
 	{
+
 		TVITEM tvItem;
 		tvItem.hItem = hItem;
 		tvItem.mask = TVIF_PARAM;
 		SendMessage(_hSelf, TVM_GETITEM, 0,(LPARAM)&tvItem);
-		if (tvItem.lParam)
-		{
-			delete (generic_string *)(tvItem.lParam);
-		}
+		if (_listener)
+			_listener->onTreeItemRemoved(hItem,(TreeViewData*)(tvItem.lParam));

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `FALSE` before using `tvItem.lParam`.